archunit-junit/junit5/engine/src/main/java/com/tngtech/archunit/junit/NamespacedStore.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2014-2022 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.junit;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+
+import com.google.common.base.Objects;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Store;
+
+@SuppressWarnings("unchecked")
+class NamespacedStore implements Store {
+
+    private final Map<NamespacedKey, Object> storage;
+    private final ExtensionContext.Namespace namespace;
+    private final Map<NamespacedKey, Object> parent;
+
+    NamespacedStore(ExtensionContext.Namespace namespace,
```

Comment:
A very simple implementation for extension storage, not sure if needed for now, haven't really tested the code without it